project/ScalazBuild.scala

```diff
@@ -115,9 +118,29 @@ object Scalaz {
           CrossType.Full.sharedSrcDir(baseDirectory.value, "main").toList.map(f => file(f.getPath + "-2.11"))
         case Some((2, x)) if x >= 12 =>
           CrossType.Full.sharedSrcDir(baseDirectory.value, "main").toList.map(f => file(f.getPath + "-2.12+"))
+        case Some((0, x)) =>
+          Seq(file(sourceDirectory.value.getPath + "/main/scala-2.12")) ++
+            CrossType.Full.sharedSrcDir(baseDirectory.value, "main").toList.map(f => file(f.getPath + "-2.12+"))
         case _ => Nil
       }
-    }
+    },
+    Test / scalacOptions ++= {
+      CrossVersion.partialVersion(scalaVersion.value) match {
+        case Some((0, _)) => Seq("-language:implicitConversions")
+        case _            => Nil
+      }
+    },
+    Test / unmanagedSourceDirectories ++= {
+      CrossVersion.partialVersion(scalaVersion.value) match {
+        case Some((0, x)) =>
+          CrossType.Full.sharedSrcDir(baseDirectory.value, "main").toList.map(f => file(f.getPath + "-2.12+"))
+        case _ => Nil
+      }
+    },
+    publishArtifact in (Compile, packageDoc) := (CrossVersion.partialVersion(scalaVersion.value) match {
```

Comment:
Maven refuses to publish artifacts without documentation jars, the usual workaround is to have an empty set of sources for which documentation should be generated:
```scala
sources in (Compile, doc) := Seq(),
```

Reply:
i am learning a LOT
thank you